src/index.js

```diff
@@ -469,7 +471,10 @@ class ReactTooltip extends React.Component {
    */
   addScrollListener (currentTarget) {
     const isCaptureMode = this.isCapture(currentTarget)
-    window.addEventListener('scroll', this.hideTooltip, isCaptureMode)
+
+    const hideTooltip = (event, hasTarget) => this.hideTooltip(event, hasTarget, { isScroll: true })
+
+    window.addEventListener('scroll', hideTooltip, isCaptureMode)
```

Comment:
Hi @aronhelser! Thanks so much for taking the time to check the PR 🙏

Ahh I see, so since `window.addEventListener('scroll', hideTooltip, isCaptureMode)` refers to a different function from `this.hideTooltip`, `window.removeEventListener('scroll', this.hideTooltip)` won't be able to clear `const hideTooltip`, and thus introducing memory leak.

Will need to think about a way around this 👍 

Thanks for catching that!